src/main/java/com/bankingservice/banking/services/servicehelper/AccountServiceHelper.java

```diff
@@ -57,32 +59,38 @@ public RegisterUserResponseDTO makeRegisterUserResponseDTO(RegisterUserModel reg
      * @param onBoardRequestDTO
      * @return userOnBoardModel
      */
-    public UserOnBoardModel makeOnBoardingEntity(OnBoardRequestDTO onBoardRequestDTO) {
+    public UserOnBoardModel makeOnBoardingEntity(OnBoardRequestDTO onBoardRequestDTO) throws UserIdNotFoundException {
         logger.info("[makeOnBoardingEntity] converting onBoardRequestDTO {} to userOnBoardModel", onBoardRequestDTO);
         UserOnBoardModel userOnBoardModel = new UserOnBoardModel();
         BeanUtils.copyProperties(onBoardRequestDTO, userOnBoardModel);
         Optional<RegisterUserModel> registerUserModel = registerUserRepository.findByUserId(onBoardRequestDTO.getUserId());
-        if(registerUserModel.isPresent()){
+        if (registerUserModel.isPresent()) {
             userOnBoardModel.setRegisterUserId(registerUserModel.get().getId());
+            return userOnBoardModel;
+        } else {
+            logger.debug("[makeOnBoardingEntity] user does not exist for userId: {}", onBoardRequestDTO.getUserId());
+            throw new UserIdNotFoundException(ErrorCode.USER_ID_NOT_FOUND, ErrorCode.USER_ID_NOT_FOUND.getErrorMessage(), ErrorCode.USER_ID_NOT_FOUND.getDisplayMessage());
         }
-        return userOnBoardModel;
```

Comment:
this return statement is not required as of above  if-else conditions, code wouldnt reach here.